src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1146,58 +1095,83 @@ private static string UpdatePath(string basePath, string pathToAdd, ref int inse
             Dbg.Assert(basePath != null, "basePath should not be null according to contract of the function");
             Dbg.Assert(pathToAdd != null, "pathToAdd should not be null according to contract of the function");
 
-            if (string.IsNullOrEmpty(pathToAdd))
+            // The 'pathToAdd' could be a 'combined path' (path-separator-separated).
+            string[] newPaths = pathToAdd.Split(
+                Path.PathSeparator,
+                StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+
+            if (newPaths.Length is 0)
             {
-                // we don't want to append empty paths.
+                // The 'pathToAdd' doesn't really contain any paths to add.
                 return basePath;
             }
 
-            StringBuilder result = new(basePath);
-            // The 'pathToAdd' could be a 'combined path' (semicolon-separated).
-            foreach (string subPathToAdd in pathToAdd.Split(Path.PathSeparator, StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries))
+            var result = new StringBuilder(basePath);
+            var addedPaths = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            string[] initialPaths = basePath.Split(
+                Path.PathSeparator,
+                StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+
+            foreach (string p in initialPaths)
             {
-                if (string.IsNullOrEmpty(subPathToAdd))
-                {
-                    // Extreme case: empty path after trimming - skip it.
-                    continue;
-                }
+                // Remove the trailing directory separators.
+                // Trailing white spaces were already removed by 'StringSplitOptions.TrimEntries'.
+                addedPaths.Add(NormalizePath(p));
+            }
 
-                // Searching in effective 'result' value ensures that possible duplicates in pathsToAdd are handled correctly.
-                if (PathContainsSubstring(result.ToString(), subPathToAdd))
+            foreach (string subPathToAdd in newPaths)
+            {
+                // Remove the trailing directory separators.
+                // Trailing white spaces were already removed by 'StringSplitOptions.TrimEntries'.
+                string normalizedPath = NormalizePath(subPathToAdd);
+                if (addedPaths.Contains(normalizedPath))
```

Comment:
We could remove line 1158 if we use Add():
```suggestion
                if (!addedPaths.Add(normalizedPath))
```